### ISSUE TITLE
Shortcuts and Sessions not working

### ISSUE THREAD

**Issue by grecodavide** [glepnir/dashboard-nvim#225]
Just installed dashboard, and I found out that these two things aren't working. I tried both with single keys shortcuts and key chords, but nothing seems to work (the default key effect gets applied).
As for sessions, the path separator gets replaced with a `_`
I tried both in neovim version 0.9 and 0.8, but these issues persist.

My config:
```lua
local has_dashboard, db = pcall(require, 'dashboard')
if not has_dashboard then
    print('dashboard not found')
    return
end


db.preview_file_path   = string.format("%s/nvim/header", os.getenv("XDG_CONFIG_HOME"))
db.preview_command     = "cat"
db.preview_file_height = 10
db.preview_file_width  = 100
db.header_pad          = 2

db.session_direcory    = string.format("%s/nvim/sessions", os.getenv("XDG_CACHE_HOME"))

db.hide_statusline     = false
db.hide_winbar         = false
db.hide_tabline        = false

db.custom_center = {
    {
        icon = '   ',
        desc = 'Oldfiles                ',
        shortcut = 'o',
        action = 'lua require("telescope.builtin").oldfiles()'
    }, {
        icon = '   ',
        desc = 'Configs                 ',
        shortcut = 'h',
        action = 'lua require("telescope").extensions.file_browser.file_browser({cwd="~/.config"})'
    }, {
        icon = '   ',
        desc = 'Binaries                ',
        shortcut = 'b',
        action = 'lua require("telescope").extensions.file_browser.file_browser({cwd="~/.bin"})'
    }, {
        icon = '   ',
        desc = 'Code                    ',
        shortcut = 'c',
        action = 'lua require("telescope").extensions.file_browser.file_browser({cwd="~/shiny-potato"})'
    },
}

db.custom_footer = {}

```

I'm on Linux, with plugins updated (dashboard 0.2.2)

**Comment by glepnir** [comment]
shortcut is the command keymap in your config. session works fine.

**Comment by grecodavide** [comment]
I don't understand your answer. The problems I have are that the shortcuts I set don't perform the actions specified (e.g. `o` doesn't open telescope oldfiles, but tries to perform the default `o` action), while sessions don't work at all: If I try to run `SessionSave name` I get this error:
![image](https://user-images.githubusercontent.com/80161991/199453715-7c3b12fd-c574-4466-990a-393aee3776e9.png)

**Comment by glepnir** [comment]
which keymap of `telescope oldfiles` ?  like `<leader>fo` the shortcuts is `<leader>fo` if you want config a buffer map `o` for command `telescope oldfiles` . you need write a `Filetype` autocmd events pattern is `dashborad` and config a buffer map in callback. note this feature I will support later a new field like `bufmap` .then you can pass `o` to it  instead of `<leader>fo` . currently you nedd use the custom way .

`SessionSave` not receive the argument. it convert the current workspace and filename . make sure your session dir can be write.

**Comment by grecodavide** [comment]
Even without arguments I get the same error. As I said before, it seems like the path separator gets replaced with an underscore (see screenshot in previous answer)
Thanks for your answers, I thought the shortcuts were automatically set

**Comment by glepnir** [comment]
> I thought the shortcuts were automatically set

no it manually.  maybe can support auto set ? 

>  it seems like the path separator gets replaced with an underscore (see screenshot in previous answer)

right it convert to underline . like i said. the error `can not open for write`  the one of reason is no permission for write . second is file name wrong? I don't think is this. 

just a min config for test on my machine  works fine
![images](https://user-images.githubusercontent.com/41671631/199461450-88712ed3-7207-490a-8a34-911f339db87d.gif)

**Comment by grecodavide** [comment]
I tried with the default config and still doesn't work. I gave writing permission to the directory but still doesn't work

**Comment by glepnir** [comment]
in my config is `/Users/joyce/.cache/nvim/session ` you can use

`lua print(require('dashboard').session_directory)`

**Comment by grecodavide** [comment]
I get an empty string, even though in my config I have set a `session_directory`

**Comment by glepnir** [comment]
typo here `db.session_direcory ` you missing `t`

**Comment by grecodavide** [comment]
Oh, you're right sorry
Thanks for your help ;)
